mypy/test/testtypes.py

```diff
@@ -451,6 +450,32 @@ def test_false_only_of_union(self) -> None:
             assert_true(fo.items[0].can_be_false)
             assert_true(fo.items[1] is tup_type)
 
+    def test_simplified_union(self) -> None:
+        fx = self.fx
+
+        self.assert_simplified_union([fx.a, fx.a], fx.a)
+        self.assert_simplified_union([fx.a, fx.b], fx.a)
+        self.assert_simplified_union([fx.a, fx.d], UnionType([fx.a, fx.d]))
+        self.assert_simplified_union([fx.a, fx.uninhabited], fx.a)
+        self.assert_simplified_union([fx.ga, fx.gs2a], fx.ga)
+        self.assert_simplified_union([fx.ga, fx.gsab], UnionType([fx.ga, fx.gsab]))
+        self.assert_simplified_union([fx.ga, fx.gsba], fx.ga)
+        self.assert_simplified_union([fx.a, UnionType([fx.d])], UnionType([fx.a, fx.d]))
+        self.assert_simplified_union([fx.a, UnionType([fx.a])], fx.a)
+        self.assert_simplified_union([fx.b, UnionType([fx.c, UnionType([fx.d])])],
+                                     UnionType([fx.b, fx.c, fx.d]))
+        self.assert_simplified_union([fx.lit1, fx.lit2], UnionType([fx.lit1, fx.lit2]))
+        self.assert_simplified_union([fx.lit1, fx.lit3], UnionType([fx.lit1, fx.lit3]))
+        self.assert_simplified_union([fx.lit1, fx.uninhabited], fx.lit1)
+        self.assert_simplified_union([fx.lit1_inst, fx.lit2_inst],
+                                     UnionType([fx.lit1_inst, fx.lit2_inst]))
+        self.assert_simplified_union([fx.lit1_inst, fx.lit3_inst],
+                                     UnionType([fx.lit1_inst, fx.lit3_inst]))
+        self.assert_simplified_union([fx.lit1_inst, fx.uninhabited], fx.lit1_inst)
+
+    def assert_simplified_union(self, original: List[Type], union: Type) -> None:
+        assert_equal(make_simplified_union(original), union)
```

Comment:
Test also `list(reversed(original))`? This way we'd get some confidence that the order of items doesn't affect the result.